test/core/api/service/chain/chain_api_test.cpp

```diff
@@ -134,9 +135,26 @@ TEST_F(ChainApiTest, GetHeaderLats) {
   EXPECT_CALL(*block_tree, getLastFinalized())
       .WillOnce(Return(BlockInfo(42, hash1)));
 
-  EXPECT_CALL(*header_repo, getBlockHeader(a))
-      .WillOnce(Return(header));
+  EXPECT_CALL(*header_repo, getBlockHeader(a)).WillOnce(Return(header));
 
   EXPECT_OUTCOME_TRUE(r, api->getHeader());
   ASSERT_EQ(r, header);
 }
+
+/**
+ * @given chain api
+ * @when call a subscribe new heads
+ * @then the correct values are returned
+ */
+TEST(StateApiTest, SubscribeStorage) {
+  auto chain_api = std::make_shared<ChainApiMock>();
+  auto p = std::dynamic_pointer_cast<ChainApi>(chain_api);
```

Comment:
Casting is really needed here? It is needless to dynamic cast from derived to base class

Reply:
Answer, aplease

Reply:
PR was not requested for review yet.

Reply:
fixed